src/contexts/ConfigContext.tsx

```diff
@@ -46,3 +29,53 @@ export function ConfigProvider({ children }: { children: any }) {
 export function useConfigContext() {
   return useContext(ConfigContext)
 }
+
+function validateStringConfig<T = string>(
+  value: string,
+  validValues: string[],
+  transformer: (value: string) => T = (value) => value as T
+) {
+  if (!validValues.includes(value)) return undefined
+  return transformer(value)
+}
+
+const latestVersion = '0.1'
+const schemaGetter = {
+  '0.1': () => {
+    const theme = getUrlQuery('theme')
+    const order = getUrlQuery('order')
+    const enableBackButton = getUrlQuery('enableBackButton')
+    const enableLoginButton = getUrlQuery('enableLoginButton')
+    const autoFocus = getUrlQuery('autoFocus')
+
+    const usedOrder = order.split(',').filter((value) => !!value)
+
+    return {
+      order: usedOrder,
+      theme: validateStringConfig(theme, ['dark', 'light']),
+      enableBackButton: validateStringConfig(
+        enableBackButton,
+        ['true', 'false'],
+        (value) => value === 'true'
+      ),
+      enableLoginButton: validateStringConfig(
+        enableLoginButton,
+        ['true', 'false'],
+        (value) => value === 'true'
+      ),
+      autoFocus: validateStringConfig(
+        autoFocus,
+        ['true', 'false'],
+        (value) => value === 'true'
+      ),
+    }
+  },
+} satisfies { [key: string]: () => State }
```

Comment:
Updated

Reply:
> Idk how the State should work when we add new version

for this one, its okay if we want to change any attribute in State type, because then we just need to update the old schema parser